tests/tpch/generate-data.py

```diff
@@ -1,47 +1,145 @@
+import functools
 import pathlib
 import tempfile
+import warnings
 
 import botocore.session
 import click
+import coiled
 import dask
 import duckdb
 import psutil
+import pyarrow.compute as pc
+from dask.distributed import LocalCluster, wait
 
 
 def generate(
-    scale: int = 10, partition_size: str = "50 MiB", path: str = "./tpch-data"
+    scale: int = 10,
+    partition_size: str = "128 MiB",
+    path: str = "./tpch-data",
+    relaxed_schema: bool = False,
 ):
     if str(path).startswith("s3"):
         path += "/" if not path.endswith("/") else ""
-        path += f"scale-{scale}/"
+        path += f"scale-{scale}{'-strict' if not relaxed_schema else ''}/"
+        use_coiled = True
     else:
-        path = pathlib.Path(path) / f"scale-{scale}/"
+        path = (
+            pathlib.Path(path)
+            / f"scale-{scale}{'-strict' if not relaxed_schema else ''}/"
+        )
         path.mkdir(parents=True, exist_ok=True)
+        use_coiled = False
 
     print(f"Scale: {scale}, Path: {path}, Partition Size: {partition_size}")
+    kwargs = dict(
+        scale=scale,
+        path=path,
+        relaxed_schema=relaxed_schema,
+        partition_size=partition_size,
+    )
+
+    if use_coiled:
+        with coiled.Cluster(
+            n_workers=10,
+            worker_memory="4 GiB",
+            worker_options={"nthreads": 1},
+            region="us-east-2",
+        ) as cluster:
+            cluster.adapt(minimum=1, maximum=250)
+            _generate_via_cluster(cluster, **kwargs)
+    else:
+        with LocalCluster(threads_per_worker=1) as cluster:
+            _generate_via_cluster(cluster, **kwargs)
+
+
+def _generate_via_cluster(cluster, scale, path, relaxed_schema, partition_size):
+    with cluster.get_client() as client:
+        jobs = []
+        for step in range(0, scale):
+            job = client.submit(
```

Comment:
no preference on my end but FYI

this can also be written as

`jobs = client.map(_tpch_data_gen, range(0, scale), ....)`

Reply:
https://github.com/coiled/benchmarks/pull/1131/commits/1344f2fb17c77de6ae4abb9f1d800687944c690b thanks!